code/game/objects/items/storage/storage.dm

```diff
@@ -30,7 +30,7 @@
 	var/list/content_watchers //list of mobs currently seeing the storage's contents
 	var/storage_flags = STORAGE_FLAGS_DEFAULT
 	var/has_gamemode_skin = FALSE ///Whether to use map-variant skins.
-
+	var/storage_holster = FALSE
```

Comment:
This is not appropriate here because `/obj/item/storage` doesn't have the concept of a holstered gun. That is a var and behavior for a `/obj/item/storage/belt/gun`.

Also it being a var is unnecessary because you can just have the proc `dump_into` overridden for either a `/obj/item/storage/belt/gun` or `/obj/item/storage/belt/gun/flaregun` to have the new behavior.

Reply:
i was thinking if i but it in /obj/item/storage/belt/gun the proc wouldn't be able to access it since it's in /obj/item/storage
also i wanted to create a new child for the proc if possible...